File: WoodPeckerDemo/WoodPeckerDemo/Modules/WebConsole/WebContentViewController.m
//
//  WebDebuggerTestViewController.m
//
//  Created by xiaogang zhang on 2017/12/17.
//  Copyright © 2017年 lifebetter. All rights reserved.
//

#import "WebContentViewController.h"

@import WebKit;
@import UIKit;

@interface WebContentViewController ()<WKUIDelegate>

@property (nonatomic, strong) UIWebView * webView;
@property (nonatomic, strong) WKWebView * wkWebView;
@property (weak, nonatomic) IBOutlet UIView *contentView;

@end

@implementation WebContentViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.navigationItem.title = self.uiwebView ? @"UIWebView" : @"WKWebView";
    [self setupWebview];
    [self loadContent];
}

- (void)setupWebview
{
    UIView * contentView = self.contentView;
    if(self.uiwebView){
        UIWebView * webView = [[UIWebView alloc] init];
        webView.frame = contentView.bounds;
        webView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
        [contentView addSubview:webView];
        self.webView = webView;
    }else{
        WKWebView * wkWebView = [[WKWebView alloc] init];
        wkWebView.UIDelegate = self;
        wkWebView.frame = contentView.bounds;
        wkWebView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
        [contentView addSubview:wkWebView];
        self.wkWebView = wkWebView;
    }
}

- (void)loadContent
{
    [[NSURLCache sharedURLCache] removeAllCachedResponses];
    NSString * url = @"http://www.woodpeck.cn";
    NSURL * requestURL = [NSURL URLWithString:url];
    NSURLRequest * request = [NSURLRequest requestWithURL:requestURL];
    if(self.uiwebView){
        [self.webView loadRequest:request];
    }else{
        [self.wkWebView loadRequest:request];
    }
}

#pragma mark -----------------   wkwebview uidelegate (Alert,Confirm support)  ----------------

- (void)webView:(WKWebView *)webView runJavaScriptAlertPanelWithMessage:(NSString *)message initiatedByFrame:(WKFrameInfo *)frame completionHandler:(void (^)(void))completionHandler
{
    if(message.length == 0) return;
    UIAlertController * alert = [UIAlertController alertControllerWithTitle:@"Alert" message:message preferredStyle:UIAlertControllerStyleAlert];
    UIAlertAction * confirm = [UIAlertAction actionWithTitle:@"Ok" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
        completionHandler();
    }];
    [alert addAction:confirm];
    [self presentViewController:alert animated:YES completion:nil];
}

- (void)webView:(WKWebView *)webView runJavaScriptConfirmPanelWithMessage:(NSString *)message initiatedByFrame:(WKFrameInfo *)frame completionHandler:(void (^)(BOOL result))completionHandler
{
    if(message.length == 0) return;
    UIAlertController * alert = [UIAlertController alertControllerWithTitle:@"Alert" message:message preferredStyle:UIAlertControllerStyleAlert];
    UIAlertAction * cancel = [UIAlertAction actionWithTitle:@"Cancel" style:UIAlertActionStyleCancel handler:^(UIAlertAction * _Nonnull action) {
        completionHandler(NO);
    }];
    UIAlertAction * confirm = [UIAlertAction actionWithTitle:@"Ok" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
        completionHandler(YES);
    }];
    [alert addAction:cancel];
    [alert addAction:confirm];
    [self presentViewController:alert animated:YES completion:nil];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}

@end

File: WoodPeckerSwiftDemo/WoodPeckerSwiftDemo/Modules/WebConsole/WebContentViewController.swift
//
//  WebContentViewController.swift
//  WoodPeckerSwiftDemo
//
//  Created by xiaogang zhang on 2018/1/21.
//  Copyright © 2018年 lifebetter. All rights reserved.
//

import UIKit
import WebKit

class WebContentViewController: UIViewController,WKUIDelegate
{

    @IBOutlet var contentView: UIView?
    var uiwebView: Bool?
    var webView: UIWebView?
    var wkWebView: WKWebView?
    
    init() {
        super.init(nibName: nil, bundle: nil)
    }
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.navigationItem.title = self.uiwebView! ? "UIWebView" : "WKWebView";
        self.setupWebview()
        self.loadContent()
    }
    
    func setupWebview(){
        let contentView: UIView! = self.contentView
        if(self.uiwebView!){
            let webView = UIWebView()
            webView.frame = contentView.bounds;
            webView.autoresizingMask = [.flexibleWidth,.flexibleHeight]
            contentView.addSubview(webView)
            self.webView = webView
        }else{
            let wkWebView = WKWebView()
            wkWebView.uiDelegate = self;
            wkWebView.frame = contentView.bounds;
            wkWebView.autoresizingMask = [.flexibleWidth,.flexibleHeight]
            contentView.addSubview(wkWebView)
            self.wkWebView = wkWebView;
        }
    }
    
    func loadContent() {
        URLCache.shared.removeAllCachedResponses()
        let url = "http://www.woodpeck.cn"
        let requestURL = URL.init(string: url)
        let request = URLRequest.init(url: requestURL!)
        if(self.uiwebView!){
            self.webView!.loadRequest(request)
        }else{
            self.wkWebView!.load(request)
        }
    }
    
/// uidelegate (Alert,Confirm support)
    func webView(_ webView: WKWebView, runJavaScriptAlertPanelWithMessage message: String, initiatedByFrame frame: WKFrameInfo, completionHandler: @escaping () -> Swift.Void)
    {
        let alert = UIAlertController.init(title: "Alert", message: message, preferredStyle: UIAlertControllerStyle.alert);
        let confirm = UIAlertAction.init(title: "Ok", style: UIAlertActionStyle.default) { (action) in
            completionHandler();
        }
        alert.addAction(confirm);
        self.present(alert, animated: true, completion: nil);
    }
    
    
    public func webView(_ webView: WKWebView, runJavaScriptConfirmPanelWithMessage message: String, initiatedByFrame frame: WKFrameInfo, completionHandler: @escaping (Bool) -> Swift.Void)
    {
        let alert = UIAlertController.init(title: "Alert", message: message, preferredStyle: UIAlertControllerStyle.alert);
        let cancel = UIAlertAction.init(title: "Cancel", style: UIAlertActionStyle.default) { (action) in
            completionHandler(false);
        }
        let confirm = UIAlertAction.init(title: "Ok", style: UIAlertActionStyle.default) { (action) in
            completionHandler(true);
        }
        alert.addAction(cancel);
        alert.addAction(confirm);
        self.present(alert, animated: true, completion: nil);
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }

}









